### PR TITLE
APIGOV-27340 - add local spec dir option

### PR DESCRIPTION
**Pull request by jcollins-axway**
- Add config to specify local path for specs
- Add ability to disable polling of apigee spec repo
- Add ability to enable publishing as unstructured
- Add extensions config for local path spec search
- Update readme for all new options
- Add /specs volume to docker container

### CODE REVIEW

**Review by vivekschauhan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

File: discovery/README.md [hunk 1]
```diff
@@ -54,19 +54,20 @@ This is the default operating mode that discoveries API Proxies and attempts to
...
+    * If local specs path set, see options below, check for the spec there using the Proxy Name as the file name and searching using the extensions
     * Proxy Revision has spec set, use it
     * Proxy Revision has association.json resource file, get path
       * Using path check to see if it is in the specs that were found by agent, use it
     * Using deployed URL path check for specs for match, use it
   * Check proxy for Key or Oauth policy for authentication
   * Create API Service
     * If spec was found use it in revision
-    * If spec was not found create as unstructured  
+    * If spec was not found create as unstructured, given option to do so is set (see below)
```
Comment (sbolosan) [inline discovery/README.md:70]: If the spec was not found,

File: discovery/README.md [hunk 2]
```diff
@@ -86,12 +87,15 @@ This mode can be setting the `APIGEE_DISCOVERYMODE` environment variable to `pro
 ### Product discovery
 
 * Find all specs
-  * Parsing is not necessary in this mode
+  * Parsing within the specs job is not necessary in this mode
   * Save info to cache
+  * If using a local specs path set the "spec_local" attribute with a value of the spec file name in that path
```
Comment (sbolosan) [inline discovery/README.md:92]: If using a local specs path,

File: discovery/README.md [hunk 3]
```diff
@@ -86,12 +87,15 @@ This mode can be setting the `APIGEE_DISCOVERYMODE` environment variable to `pro
...
 * Find all Products defined
   * Using the `APIGEE_FILTER` determine if the product should be discovered
-  * Using the product's name or display name, match it to a spec (case insensitive)
-  * If a spec is found create an API Service
+  * Determine spec
+    * Is a spec_local attribute is set on the product look for the spec in the local specs path
```
Comment (sbolosan) [inline discovery/README.md:96]: If a spec_local? Instead of Is a spec_local

File: discovery/README.md [hunk 4]
```diff
@@ -86,12 +87,15 @@ This mode can be setting the `APIGEE_DISCOVERYMODE` environment variable to `pro
...
+    * Using the product's name or display name, match it to a spec (case insensitive)
+  * If a spec is found create an API Service (create as unstructured when no spec is found, if optino set)
```
Comment (sbolosan) [inline discovery/README.md:98]: If a spec is found,

**Review by sbolosan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]
when you can trim, trim 👍

**Pull request merged**